connector/connect/client/jvm/pom.xml

```diff
@@ -125,6 +125,11 @@
       <version>${mima.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
```

Comment:
I have trouble understanding how adding this to the client influences the server to do the right thing? Could you help me understand?

cc @zhenlineo

Reply:
Good question, when I add this to server side, the test still failed, I also think this is strange, let me think more.

Reply:
Yeah, even if I make `parquet-hadoop` as the compile dependency of the `server` module(but exclude `parquet-` from the SBT assembly), this test will still fail.

So I think this issue is  is related to the startup class path used for `SimpleSparkConnectService`.

Reply:
Did an experiment

https://github.com/apache/spark/blob/eb24dbd305b9e46382d49d43202d0c3a89fd324f/connector/connect/server/src/main/scala/org/apache/spark/sql/connect/SimpleSparkConnectService.scala#L40-L44

`System.getProperty("java.class.path")` before `while (true) {` and print the results:  [SimpleSparkConnectService-classpath.txt](https://github.com/apache/spark/files/10811046/SimpleSparkConnectService-classpath.txt), there is no jar related to `parquet`

Reply:
And when I revert to add `parquet-hadoop` as test  dependency of `connect-client-jvm`, I found the `parquet-**.jar` in the `java.class.path`, so I think `SimpleSparkConnectService` startup use classpath of `connect-client-jvm` now in this scenario.

Reply:
So I think `parquet-hadoop` is added to `connect-client-jvm` module instead of `connect` server module because `SimpleSparkConnectService` is started using the test classpath of `connect-client-jvm` module currently .

Reply:
Or do you have any suggestions to correct the startup classpath of `SimpleSparkConnectService`? @hvanhovell @zhenlineo

Reply:
I kind of understand the `SimpleSparkConnectService` started the test with the client classpath, but then the submit script should scan all spark dependencies independently. See this code in spark-class:
https://github.com/apache/spark/blob/master/bin/spark-class#L38-L43
These are the real jars that are pulled into the spark when starting. So I feel we should ensure the jars are put into these jar folders correctly instead of modifying the client.

Reply:
both `${SPARK_HOME}/assembly/target/scala-$SPARK_SCALA_VERSION/jars` and `SPARK_JARS_DIR="${SPARK_HOME}/jars"` not exists. the new commit make `${SPARK_HOME}/assembly/target/scala-$SPARK_SCALA_VERSION/jars`  has content.
But this makes the module test more heavier ...

Reply:
Do we still need rules for the following two jars?
```
      (LocalProject("connect") / assembly).value
      (LocalProject("connect-client-jvm") / assembly).value
```
These two jars are not included in `${SPARK_HOME}/assembly/target/scala-$SPARK_SCALA_VERSION/jars`.

Is it slow for you to run sbt package to run this test? I get the impression it takes ~1min to run the first time? On the other side, maven is so slow for me.

Reply:
> Do we still need rules for the following two jars?
> 
> ```
>       (LocalProject("connect") / assembly).value
>       (LocalProject("connect-client-jvm") / assembly).value
> ```
> 
> These two jars are not included in `${SPARK_HOME}/assembly/target/scala-$SPARK_SCALA_VERSION/jars`.
> 
> Is it slow for you to run sbt package to run this test? I get the impression it takes ~1min to run the first time? On the other side, maven is so slow for me.

No longer needed, `sql`, `assembly ` will trigger their build

Reply:
Update comments

Reply:
> On the other side, maven is so slow for me.

Yes, maven is slow, but we must also ensure `maven pack test` can pass without any preparation work, otherwise, they will become a block of new version release :)